src/Microsoft.AspNetCore.Mvc.Core/Internal/CompositeChangeToken.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    internal class CompositeChangeToken : IChangeToken
```

Comment:
There's a `CompositeFileChangeToken` in FileSystem but not just a vanilla `CompositeChangeToken`. I added a work item to figure out if we can commonize it - https://github.com/aspnet/Common/issues/163